### PR TITLE
fix: restoring KVBM wheel generation for `framework=none` and add CI validation 

### PR DESCRIPTION
**Pull request by oandreeva-nv**
#### Overview:

This PR fixes KVBM wheel generation when using `--enable-kvbm` with `--framework none` and adds comprehensive CI tests to validate KVBM wheels are built correctly across all frameworks.

#### Details:

Problem                                                                                                                                                                                  

Previously, the build script would ignore the --enable-kvbm flag when using --framework none, unconditionally setting ENABLE_KVBM=false for non-VLLM/TRTLLM frameworks. This prevented users from building KVBM wheels in base images without a specific framework.                                                                                                   

Additionally, there were no CI tests to catch regressions in KVBM wheel generation.                                                                                                      

Changes                                                                                                                                                                                  
                                                                                                                                                                                           
  1. Fixed build script logic (container/build.sh)                                                                                                                                         
                                                                                                                                                                                           
  - Added explicit default: Set ENABLE_KVBM=false at script initialization for clarity (line 130)                                                                                          
  - Fixed flag handling: Modified framework logic to respect user's explicit --enable-kvbm flag                                                                                            
    - VLLM/TRTLLM: Force ENABLE_KVBM=true (existing behavior)                                                                                                                              
    - Other frameworks: Respect user's flag or default to false (fixed behavior)                                                                                                           
  - Simplified logic: Only pass build arg when ENABLE_KVBM == "true" (more idiomatic and efficient)                                                                                        
                                                                                                                                                                                           
  2. Added comprehensive pytest-based CI validation       
  New test file (tests/dependencies/test_kvbm_imports.py):                                                                                                                                 
  - 2 helper functions for reusable verification logic:
    - _check_kvbm_wheel_exists() - Verifies wheel file exists at /opt/dynamo/wheelhouse/kvbm*.whl
    - _check_kvbm_imports() - Verifies KVBM package and core classes can be imported
  - 6 test functions marked with @pytest.mark.pre_merge across 3 contexts:
    - Base tests (no framework markers) - Run in main job with --framework none --enable-kvbm:
        - test_kvbm_wheel_exists()
      - test_kvbm_imports()
    - vLLM tests (@pytest.mark.vllm) - Run in vLLM job:
        - test_kvbm_wheel_exists_vllm()
      - test_kvbm_imports_vllm()
    - TRT-LLM tests (@pytest.mark.trtllm) - Run in TRT-LLM job:
        - test_kvbm_wheel_exists_trtllm()
      - test_kvbm_imports_trtllm()

  Updated main build (container-validation-dynamo.yml):
  - Modified build command to include --enable-kvbm flag
  - KVBM tests automatically run as part of existing pytest suite
  - Seamlessly integrated into main "Build and Test - dynamo" workflow                                         

#### Where should the reviewer start?

<!-- call out specific files that should be looked at closely -->

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)

- closes GitHub issue: #xxx


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Chores**
  * Enhanced container validation workflows to verify KVBM wheel presence and functionality in built images.
  * Updated build scripts to support KVBM feature enablement for vLLM and TRT-LLM frameworks.

* **Tests**
  * Added new test job to validate KVBM wheel generation and import capabilities.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

**Comment by oandreeva-nv** [comment]
/ok to test [df5935b](https://github.com/ai-dynamo/dynamo/pull/4772/commits/df5935b5f74a21f5519270a34dc518782cc79aeb)

File: .github/workflows/container-validation-dynamo.yml [hunk 1]
```diff
@@ -97,6 +97,46 @@ jobs:
...
+          echo "${{ secrets.NGC_CI_ACCESS_TOKEN }}" | docker login nvcr.io -u '$oauthtoken' --password-stdin
+      - name: Define Image Tag
+        id: define_image_tag
+        run: |
+          echo "image_tag=dynamo:kvbm-test" >> $GITHUB_OUTPUT
+      - name: Build image with KVBM enabled
+        env:
+          GITHUB_TOKEN: ${{ secrets.CI_TOKEN }}
+        run: |
+          ./container/build.sh --tag ${{ steps.define_image_tag.outputs.image_tag }} --target dev --framework none --enable-kvbm
```
Comment (nv-kmcgill53) [inline .github/workflows/container-validation-dynamo.yml]: The KVBM build doesn't change from framework to framework, i.e. the kvbm build for trtllm/vllm/none should be the same wheel. Instead of having to build the entire image, could we use the `--dry-run` flag to print out the configuration and then assert that `ENABLE_KVBM==true`? 

For the functional installation tests, we should be able to lump this into the framework builds such that we test the installation of the wheel and the `import`ing of modules with those containers which we build anyway. Let me know if I have forgotten a corner case.

Comment (oandreeva-nv) [reply .github/workflows/container-validation-dynamo.yml]: previous issue was due to overriding `ENABLE_KVBM` value inside the build script, I don't think that `--dry-run` is a safe approach tbh. Plus, in docs, whe specify that users can build kvbm wheels with `--framework none`, so I'd like to test that

Comment (oandreeva-nv) [reply .github/workflows/container-validation-dynamo.yml]: plus we can re-use this dedicated job in future, when we migrate tests

Comment (oandreeva-nv) [reply .github/workflows/container-validation-dynamo.yml]: migrated dedicated workflow to existing one

**Review by nv-kmcgill53 — APPROVED** [review]

File: .github/workflows/container-validation-backends.yml [hunk 2]
```diff
@@ -179,6 +179,14 @@ jobs:
           azure_acr_user: ${{ secrets.AZURE_ACR_USER }}
           azure_acr_password: ${{ secrets.AZURE_ACR_PASSWORD }}
 
+      - name: Verify KVBM wheel exists in vLLM image
```
Comment (nv-anants) [inline .github/workflows/container-validation-backends.yml]: can remove these as well? run as part of pytest please

Comment (oandreeva-nv) [reply .github/workflows/container-validation-backends.yml]: moved to pytest

**Comment by oandreeva-nv** [comment]
/ok to test [7a22b96](https://github.com/ai-dynamo/dynamo/pull/4772/commits/7a22b962ddb58fa7a026234847c63cc6f409620f)

**Comment by oandreeva-nv** [comment]
/ok to test [739a02e](https://github.com/ai-dynamo/dynamo/pull/4772/commits/739a02eb9d5ef6bba7071096c042add6d745daeb)

**Review by atchernych — APPROVED** [review]

**Comment by oandreeva-nv** [comment]
/ok to test [d480a9d](https://github.com/ai-dynamo/dynamo/pull/4772/commits/d480a9d6471cd9f9dd9c67db9c8b02b127350c60)
